govcd/vm_test.go

```diff
@@ -985,23 +987,92 @@ func (vcd *TestVCD) Test_GetVirtualHardwareSection(check *C) {
 	vm, err := vcd.client.Client.GetVMByHref(existingVm.HREF)
 	check.Assert(err, IsNil)
 
-	// Preform check of virtual hardware section
-	section, err := vm.GetVirtualHardwareSection()
+	storageProfile, err := vcd.vdc.FindStorageProfileReference(vcd.config.VCD.StorageProfile.SP1)
+	check.Assert(err, IsNil)
+	isThinProvisioned := true
+	iops := int64(0)
+
+	diskSettings := &types.DiskSettings{
+		SizeMb:            1024,
+		UnitNumber:        0,
+		BusNumber:         1,
+		AdapterType:       "6",
+		ThinProvisioned:   &isThinProvisioned,
+		StorageProfile:    &storageProfile,
+		OverrideVmDefault: true,
+		Iops:              &iops,
+	}
+	diskId, err := vm.AddInternalDisk(diskSettings)
+
+	check.Assert(err, IsNil)
+	check.Assert(diskId, NotNil)
+
+	//verify
+	err = vm.Refresh()
 	check.Assert(err, IsNil)
 
-	// Check that section.Info is not Nil, as its the only field that may not be omitted when marshalled
-	check.Assert(section.Info, NotNil)
+	disk, err := vm.GetInternalDiskById(diskId, true)
+	check.Assert(err, IsNil)
 
-	// Check that section.Item is not Nil before looping over it
-	check.Assert(section.Item, NotNil)
+	check.Assert(disk.StorageProfile.HREF, Equals, storageProfile.HREF)
+	check.Assert(disk.AdapterType, Equals, diskSettings.AdapterType)
+	check.Assert(*disk.ThinProvisioned, Equals, *diskSettings.ThinProvisioned)
+	check.Assert(*disk.Iops, Equals, *diskSettings.Iops)
+	check.Assert(disk.SizeMb, Equals, diskSettings.SizeMb)
+	check.Assert(disk.UnitNumber, Equals, diskSettings.UnitNumber)
+	check.Assert(disk.BusNumber, Equals, diskSettings.BusNumber)
+	check.Assert(disk.AdapterType, Equals, diskSettings.AdapterType)
+}
 
-	// Loop over the Items to ensure
-	for _, item := range section.Item {
-		check.Assert(item.ResourceType, NotNil)
-		check.Assert(item.ResourceSubType, NotNil)
+// Test delete internal disk For VM
+func (vcd *TestVCD) Test_DeleteInternalDisk(check *C) {
```

Comment:
There is a lot in common between this function and `Test_AddInternalDisk`. Can you combine them?
If not, could you add a creation function that can be called by both?

Reply:
done